src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareFunction(a, b) {
+    const aStr = String(a);
+    const bStr = String(b);
+
+    if (aStr > bStr) {
+      return 1;
+    }
+
+    if (aStr < bStr) {
+      return -1;
+    }
+
+    return 0;
+  }
+
+  [].__proto__.sort2 = function (customCompareFunction = compareFunction) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (customCompareFunction(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
+
+    return this;
```

Comment:
The bubble sort algorithm used here is a valid and simple approach. However, it is known for its inefficiency on large datasets due to its O(n^2) complexity. You might consider implementing a more efficient sorting algorithm like quicksort or mergesort for better performance on larger arrays.